tests/unit/src/test/scala/tests/SemanticHighlightLspSuite.scala

```diff
@@ -0,0 +1,230 @@
+package tests
+
+import munit.TestOptions
+
+/**
+ * Test for request "textDocument/semanticTokens/full"
+ * https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens
+ */
+class SemanticHighlightLspSuite extends BaseLspSuite("SemanticHighlight") {
+
+  check(
+    "class, object, var, val(readonly), method, type, parameter, String(single-line)",
+    s"""|<<class>>/*keyword*/  <<Test>>/*class*/{
+        |
+        | <<var>>/*keyword*/ <<wkStr>>/*variable*/ = <<"Dog-">>/*string*/
+        | <<val>>/*keyword*/ <<nameStr>>/*variable,readonly*/ = <<"Jack">>/*string*/
+        |
+        | <<def>>/*keyword*/ <<Main>>/*method*/={
+        |
+        |  <<val>>/*keyword*/ <<preStr>>/*variable,readonly*/= <<"I am ">>/*string*/
+        |  <<var>>/*keyword*/ <<postStr>>/*variable*/= <<"in a house. ">>/*string*/
+        |  <<wkStr>>/*variable*/=<<nameStr>>/*variable,readonly*/ <<+>>/*method*/ <<"Cat-">>/*string*/
+        |
+        |  <<testC>>/*class*/.<<bc>>/*method*/(<<preStr>>/*variable,readonly*/
+        |    <<+>>/*method*/ <<wkStr>>/*variable*/
+        |    <<+>>/*method*/ <<preStr>>/*variable,readonly*/)
+        | }
+        |}
+        |
+        |<<object>>/*keyword*/  <<testC>>/*class*/{
+        |
+        | <<def>>/*keyword*/ <<bc>>/*method*/(<<msg>>/*parameter*/:<<String>>/*type*/)={
+        |   <<println>>/*method*/(<<msg>>/*parameter*/)
+        | }
+        |}
+        |""".stripMargin,
+  )
+
+  check(
```

Comment:
Let's also add some tests to cross, I will do a quick PR to your branch tomorrow.